### ISSUE TITLE
Adding `jax.nn.activations.log_cosh`

### ISSUE THREAD

**Issue by PhilipVinc** [google/jax#7245]
Would you accept new activation functions such as `log_cosh` ?
Tensorflow/keras/pytorch all have it.
The implementation is not trivial (though simple enough) if one wants to have good numerical precision.

If you'd accept it i'll open a PR

```python
def log_cosh(x):
    sgn_x = -2 * jnp.signbit(x.real) + 1
    x = x * sgn_x
    return x + jnp.log1p(jnp.exp(-2.0 * x)) - jnp.log(2.0)
```

**Comment by hawkinsp** [comment]
Keras version: https://www.tensorflow.org/api_docs/python/tf/keras/losses/log_cosh (N.B. it appears to be a loss function?)
I couldn't see a PyTorch version?

**Comment by hawkinsp** [comment]
There was no follow-up on this issue; closing. Feel free to reopen if you think this function would be of general interest, although I'm not seeing such an activation function in PyTorch or TF.
